src/bluesky/bundlers.py

```diff
@@ -717,26 +749,28 @@ async def _describe_collect(self, collect_object: Flyable):
         await self._ensure_cached(collect_object, collect=True)
 
         describe_collect = self._describe_collect_cache[collect_object]
-        describe_collect_items = list(self._maybe_format_datakeys_with_stream_name(describe_collect))
+        describe_collect_items = self._format_datakeys_with_stream_name(describe_collect)
 
-        local_descriptors: dict[Any, dict[frozenset[str], ComposeDescriptorBundle]] = {}
+        local_descriptors: dict[frozenset[str], ComposeDescriptorBundle] = {}
 
         # Check that singly nested stuff should have been pre-declared
         def is_data_key(obj: Any) -> bool:
             return isinstance(obj, dict) and {"dtype", "shape", "source"}.issubset(frozenset(obj.keys()))
 
         assert all(not is_data_key(value) for value in describe_collect.values()), (
-            "Single nested data keys should be pre-decalred"
+            "Single nested data keys should be pre-declared"
         )
 
-        # Make sure you can't use identidal data keys in multiple streams
-        duplicates: dict[str, DataKey] = defaultdict(dict)
-        for stream, data_keys in describe_collect.items():
-            for key, stuff in data_keys.items():
-                for other_stream, other_data_keys in describe_collect.items():
-                    for other_key, other_stuff in other_data_keys.items():
-                        if stream != other_stream and key == other_key and stuff == other_stuff:
-                            duplicates[stream][key] = stuff
+        # Make sure you can't use identical data keys in multiple streams
+        # TODO: Does `DataKey` have a `__eq__` method implemented?
+        # Data structure is assumed to be dict[stream_name, dictionary of data_keys]
+        duplicates: dict[str, dict[str, DataKey]] = defaultdict(lambda: {})
+        for stream, data_keys in describe_collect_items:
+            for name, data_key in data_keys.items():
+                for other_stream, other_data_keys in describe_collect_items:
+                    for other_name, other_data_key in other_data_keys.items():
+                        if stream != other_stream and name == other_name and data_key == other_data_key:
+                            duplicates[stream][name] = data_key
         if len(duplicates) > 0:
             raise RuntimeError(
                 f"Can't use identical data keys in multiple streams: {duplicates}",
```

Comment:
This'll also print the whole dict in the first line. Should probably take away 

```suggestion
                f"Can't use identical data keys in multiple streams:",
```